From c89939f410f31255f2dad21c1ed4f160f85d743c Mon Sep 17 00:00:00 2001
From: Hans Petter Selasky <hps@selasky.org>
Date: Sun, 5 Apr 2020 11:43:20 +0200
Subject: [PATCH 1519/1936] Only set initial oldamplitude when creating a
 SUBnote. Currently the oldamplitude was also set on all note velocity
 updates.

Signed-off-by: Hans Petter Selasky <hps@selasky.org>
---
 src/Synth/SUBnote.cpp | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/src/Synth/SUBnote.cpp b/src/Synth/SUBnote.cpp
index 1b243e9d..10e54fd9 100644
--- a/src/Synth/SUBnote.cpp
+++ b/src/Synth/SUBnote.cpp
@@ -202,8 +202,6 @@ void SUBnote::setup(float velocity_,
                                       pars.PGlobalFilterVelocityScaleFunction);
         }
     }
-
-    oldamplitude = newamplitude;
 }
 
 SynthNote *SUBnote::cloneLegato(void)
@@ -397,6 +395,7 @@ void SUBnote::initparameters(float freq, WatchManager *wm, const char *prefix)
         GlobalFilter->addMod(*GlobalFilterEnvelope);
     }
     computecurrentparameters();
+    oldamplitude = newamplitude;
 }
 
 /*
-- 
2.47.0

